python/ray/tune/suggest/util.py

```diff
@@ -29,3 +29,26 @@ def set_search_properties_backwards_compatible(
             return set_search_properties_func(metric, mode, config)
         else:
             raise e
+
+
+def scheduler_set_search_properties_backwards_compatible(
+    set_search_properties_func, metric: Optional[str], mode: Optional[str], **spec
+) -> bool:
+    """Wraps around set_search_properties() so that it is backward compatible.
+
+    Also outputs a warning to encourage customm searchers to be updated.
+    """
+    try:
+        return set_search_properties_func(metric, mode, **spec)
+    except TypeError as e:
+        if str(e).startswith(
+            "set_search_properties() got an unexpected keyword argument"
+        ):
+            logger.warning(
+                "Please update custom Scheduler to take in function signature "
+                "as ``def set_search_properties(metric, mode, "
+                "**spec) -> bool``."
+            )
+            return set_search_properties_func(metric, mode)
+        else:
+            raise e
```

Comment:
Can we use `inspect.signature` and `signature.bind()` instead?

Reply:
Sure. Should the other function be also changed for consistency?